packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -930,6 +938,27 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         }
     }
 
+    /**
+     * Retrieves the runtime for a data store if it's referenced as per the initially summary that it is loaded with.
+     * This is a workaround to handle scenarios where a data store shared with an external app is deleted and marked
+     * as unreferenced by GC.
+     * @param id - Id supplied during creating the data store.
+     * @param wait - True if you want to wait for it.
+     * @returns the data store runtime if the data store exists and is initially referenced; undefined otherwise.
+     */
+    private async getDataStoreIfInitiallyReferenced(id: string, wait = true): Promise<IFluidRouter> {
+        const dataStoreContext = await this.dataStores.getDataStore(id, wait);
+        // The data store is referenced if used routes in the initial summary has a route to self.
```

Comment:
Is this long-term approach? I.e. once we get flags back from snapshot, should we use them?

Reply:
I am not sure. If we don't have to get initial snapshot for that, it might be more efficient to use it. I think we can decide once we have that and see which one makes more sense.